File: src/main/java/com/massivedisaster/kickoff/config/ProjectConfiguration.java
package com.massivedisaster.kickoff.config;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.HashMap;
import java.util.List;

public class ProjectConfiguration {

	@JsonProperty("template")
	private String template;
	@JsonProperty("language")
	private String language;
	@JsonProperty("gradlePluginVersion")
	private String gradlePluginVersion;
	@JsonProperty("gradleVersion")
	private String gradleVersion;
	@JsonProperty("projectName")
	private String projectName;
	@JsonProperty("packageName")
	private String packageName;
	@JsonProperty("minimumSdkApi")
	private int minimumSdkApi;
	@JsonProperty("targetSdkApi")
	private int targetSdkApi;
	@JsonProperty("buildTools")
	private String buildTools;
	@JsonProperty("manifestKeys")
	private HashMap<String, HashMap<String, String>> manifestKeys;
	@JsonProperty("extraBuildVars")
	private HashMap<String, HashMap<String, String>> extraBuildVars;
	@JsonProperty("network")
	private Network network;
	@JsonProperty("dependencies")
	private List<Dependency> dependencies;
	@JsonProperty("hasQa")
	private boolean hasQa;
	@JsonProperty("useAndroidX")
	private boolean useAndroidX;
	@JsonProperty("hasOneSignal")
	private boolean hasOneSignal;
	@JsonProperty("slackChannel")
	private String slackChannel;
	@JsonProperty("testFairyKey")
	private String testFairyKey;
	@JsonProperty("kotlinVersion")
	private String kotlinVersion;
	@JsonProperty("slackSlug")
	private String slackSlug;

	public String getTemplate() {
		return template;
	}

	public String getLanguage() {
		return language;
	}

	public String getGradlePluginVersion() {
		return gradlePluginVersion;
	}

	public String getGradleVersion() {
		return gradleVersion;
	}

	public String getProjectName() {
		return projectName;
	}

	public String getPackageName() {
		return packageName;
	}

	public int getMinimumSdkApi() {
		return minimumSdkApi;
	}

	public int getTargetSdkApi() {
		return targetSdkApi;
	}

	public String getBuildTools() {
		return buildTools;
	}

	public HashMap<String, HashMap<String, String>> getManifestKeys() {
		return manifestKeys;
	}

    public HashMap<String, HashMap<String, String>> getExtraBuildVars() {
        return extraBuildVars;
    }

	public Network getNetwork() {
		return network;
	}

	public List<Dependency> getDependencies() {
		return dependencies;
	}

	public boolean getHasQa() {
		return hasQa;
	}

	public boolean getUseAndroidX() {
		return useAndroidX;
	}

	public boolean getHasOneSignal() {
		return hasOneSignal;
	}

	public String getSlackChannel() {
		return slackChannel;
	}

	public String getTestFairyKey() {
		return testFairyKey;
	}

	public String getKotlinVersion() {
		return kotlinVersion;
	}

	public String getSlackSlug() {
		return slackSlug;
	}
}

File: src/main/java/com/massivedisaster/kickoff/config/ProjectConfiguration.kt
package com.massivedisaster.kickoff.config

import com.fasterxml.jackson.annotation.JsonProperty

data class ProjectConfiguration(
    @JsonProperty("template") val template: String,
    @JsonProperty("language") val language: String,
    @JsonProperty("gradlePluginVersion") val gradlePluginVersion: String,
    @JsonProperty("projectName") val projectName: String,
    @JsonProperty("packageName") val packageName: String,
    @JsonProperty("minimumSdkApi") val minimumSdkApi: Int,
    @JsonProperty("targetSdkApi") val targetSdkApi: Int,
    @JsonProperty("gradleVersion") val gradleVersion: String,
    @JsonProperty("kotlinVersion") val kotlinVersion: String,
    @JsonProperty("benManesVersion") val benManesVersion: String,
    @JsonProperty("grGitVersion") val grGitVersion: String,
    @JsonProperty("network") val network: Network,
    @JsonProperty("manifestKeys") val manifestKeys: HashMap<String, HashMap<String, String>>?,
    @JsonProperty("extraBuildVars") val extraBuildVars: HashMap<String, HashMap<String, String>>?,
    @JsonProperty("dependencies") val dependencies: List<Dependency>,
    @JsonProperty("hasQa") val hasQa: Boolean,
    @JsonProperty("hasOneSignal") val hasOneSignal: Boolean,
    @JsonProperty("oneSignal") val oneSignal: OneSignal?,
    @JsonProperty("hasFirebase") val hasFirebase: Boolean,
    @JsonProperty("firebase") val firebase: Firebase?,
    @JsonProperty("hasDaggerHilt") val hasDaggerHilt: Boolean,
    @JsonProperty("daggerHilt") val daggerHilt: DaggerHilt?
)

data class Network(
    @JsonProperty("timeout") val timeout: Int,
    @JsonProperty("endpoints") val endpoints: HashMap<String, String>
)

data class Dependency(
    @JsonProperty("name") val name: String,
    @JsonProperty("globalVersion") val globalVersion: String,
    @JsonProperty("list") val list: HashMap<String, LibraryModule>
)

data class LibraryModule(
    @JsonProperty("group") val group: String,
    @JsonProperty("version") val version: String?,
    @JsonProperty("isCompiler") val compiler: Boolean
)

data class OneSignal(
    @JsonProperty("version") val version: String,
    @JsonProperty("dependencies") val dependencies: HashMap<String, LibraryModule>
)

data class Firebase(
    @JsonProperty("version") val version: String,
    @JsonProperty("distribution") val distribution: Distribution,
    @JsonProperty("classpaths") val classpaths: List<LibraryModule>,
    @JsonProperty("dependencies") val dependencies: HashMap<String, String>
)

data class Distribution(
    @JsonProperty("ids") val ids: HashMap<String, String>,
    @JsonProperty("groups") val groups: List<String>
)

data class DaggerHilt(
    @JsonProperty("version") val version: String,
    @JsonProperty("plugin") val plugin: String,
    @JsonProperty("dependencies") val dependencies: HashMap<String, LibraryModule>
)
